app/src/test/java/com/amaze/filemanager/activities/ShareContentIntentTest.java

```diff
@@ -0,0 +1,158 @@
+package com.amaze.filemanager.activities;
+
+import android.app.Dialog;
+import android.content.ClipData;
+import android.content.ClipDescription;
+import android.content.ContentResolver;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Environment;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import com.afollestad.materialdialogs.DialogAction;
+import com.afollestad.materialdialogs.MaterialDialog;
+import com.amaze.filemanager.BuildConfig;
+import com.amaze.filemanager.R;
+import com.amaze.filemanager.test.ShadowContentResolverWithLocalOutputStreamSupport;
+
+import org.apache.commons.compress.utils.IOUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.Robolectric;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.Shadows;
+import org.robolectric.annotation.Config;
+import org.robolectric.shadow.api.Shadow;
+import org.robolectric.shadows.ShadowDialog;
+import org.robolectric.shadows.ShadowToast;
+import org.robolectric.shadows.multidex.ShadowMultiDex;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import jahirfiquitiva.libs.fabsmenu.FABsMenu;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(constants = BuildConfig.class, shadows = {ShadowMultiDex.class, ShadowContentResolverWithLocalOutputStreamSupport.class})
+public class ShareContentIntentTest {
+
+    private static final String content = "Test content";
+
+    private static final Uri uri = Uri.parse("content://foo.bar.test.streamprovider/temp/thisisatest.txt");
+
+    private MainActivity activity;
+
+    @Before
+    public void setUp() throws IOException {
+        activity = Robolectric.setupActivity(MainActivity.class);
+        activity.onResume();
```

Comment:
Why are you explicitly calling `onResume()`? I assume `setupActivity()` would call `resume()`, that would be the correct code for resuming the activity.